### PR TITLE
[ascii] Add Aptos Ascii subcommand with logo command

### PR DESCRIPTION
**Pull request by perryjrandall**
Add aptos ascii art to cli (say that three times fast)

This is something I made on a whim a while ago and with Zach's prompting wanna add to share with our community!

Test Plan: running locally

```
cargo run -p aptos ascii logo 
    Finished dev [unoptimized + debuginfo] target(s) in 0.81s
     Running `target/debug/aptos ascii logo`
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMWNNXKK0000KKXXNWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMWNKkdl:;'............';:ldkKNWMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMWXOo:'.                        .':oOXWMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMWKd:.                                  .:dKWMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMXx:.                                        .:xXWMMMMMMMMMMMMMMM
MMMMMMMMMMMMMWKo'                                      ...     'oKWMMMMMMMMMMMMM
MMMMMMMMMMMMXo.                                      ,xXXO:.     .oKMMMMMMMMMMMM
MMMMMMMMMMW0:......................................;xNMMMMNk;......:OWMMMMMMMMMM
MMMMMMMMMMWXKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK0KNMMMMMMMMWXKKKKKKXWMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMW0xkXWMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMW0c.  'dXMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMN0OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOxc.      ,dOOOOOOOOOOOOOOOOOOOO0NMMMMM
MMMMNl.                                                                  .lNMMMM
MMMMO.                                     .                              .kMMMM
MMMNl                                  ..:k0k:.                            lNMMM
MMMK;                                .:OXWMMMWk;.                          ;KMMM
MMMN0kkkkkkkkkkkkkkkkkkkkkkkkkkkkxkkk0NMMMMMMMMN0kkkkkkkkkkkkkkkkkkkkkkkkkk0NMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMN00XWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNk;..'dNMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMWXXXXXXXXXXXXXXXXXXXXXXXKk:.     ,xKXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXNMMMM
MMMMk,.......................          ...................................,kMMMM
MMMMK,                                                                    ,KMMMM
MMMMWx.                    .cdo,                                         .xWMMMM
MMMMMNl                  .lKWMMXd'                                       lNMMMMM
MMMMMMNxlllllllllllllllldKWMMMMMMXxlllllllllllllllllllllllllllllllllllllxXMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMWXkxKWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMXd'  .lKWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMNOo,      .ok000000000000000000000000000000000000000KNMMMMMMMMMMMMM
MMMMMMMMMMMMMNd'                                               .,xNMMMMMMMMMMMMM
MMMMMMMMMMMMMMWXd,.                                           ,dXWMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMNOl'.                                    .'lONMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMN0d:.                              .:oONMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMWX0dl;'.                  .';ldOXWMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMWNK0kxdollccccllodxk0KNWMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
                        ___  ______ _____ _____ _____
                       / _ \ | ___ \_   _|  _  /  ___|
                      / /_\ \| |_/ / | | | | | \ `--.
                      |  _  ||  __/  | | | | | |`--. \
                      | | | || |     | | \ \_/ /\__/ /
                      \_| |_/\_|     \_/  \___/\____/

       
```

### CODE REVIEW

**Review by banool — APPROVED** [review]
Fun!!

Later on we could make ascii a subcommand group and have different subcommands in it, eg logo. But that can come later.

**Review by 0xjinn — APPROVED** [review]

File: crates/aptos/Cargo.toml [hunk 1]
```diff
@@ -62,6 +62,7 @@ diesel-async = { workspace = true }
 dirs = { workspace = true }
 futures = { workspace = true }
 hex = { workspace = true }
+indoc = { workspace = true }
```
Comment (gregnazario) [inline crates/aptos/Cargo.toml:65]: Can we do this without adding `indoc` as a dependency, or is it necessary?

File: crates/aptos/src/ascii/mod.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+/// Aptos Ascii Art
+///
+/// This tool is used to celebrate aptos in spectacular ascii
+/// fashion!
+#[derive(Debug, Parser)]
+pub struct AsciiTool {}
+
+impl AsciiTool {
+    pub async fn execute(self) -> CliResult {
```
Comment (gregnazario) [inline crates/aptos/src/ascii/mod.rs:16]: This tool doesn't implement the required interfaces to be the same as other commands in the tool.

**Review by gregnazario — CHANGES_REQUESTED** [review]
Where's the test plan and the example?

I suspect this will print out into a JSON blob which won't be fun, probably want to print out separately of the JSON.

File: crates/aptos/Cargo.toml (hunk 1 above)
Comment (perryjrandall) [reply crates/aptos/Cargo.toml:65]: I mean.... its pretty nice to have and its a very simple dependency, i find its generally useful and probably already included transitively

File: crates/aptos/src/ascii/mod.rs (hunk 2 above)
Comment (perryjrandall) [reply crates/aptos/src/ascii/mod.rs:16]: What is the required interface?

This has the async execute and there was no explicit interface I could find?

**Comment by perryjrandall** [comment]
> Where's the test plan and the example?

Sure I can add

> I suspect this will print out into a JSON blob which won't be fun, probably want to print out separately of the JSON.

No it prints the correct thing

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.81s
     Running `target/debug/aptos ascii`
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMWNNXKK0000KKXXNWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMWNKkdl:;'............';:ldkKNWMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMWXOo:'.                        .':oOXWMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMWKd:.                                  .:dKWMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMXx:.                                        .:xXWMMMMMMMMMMMMMMM
MMMMMMMMMMMMMWKo'                                      ...     'oKWMMMMMMMMMMMMM
MMMMMMMMMMMMXo.                                      ,xXXO:.     .oKMMMMMMMMMMMM
MMMMMMMMMMW0:......................................;xNMMMMNk;......:OWMMMMMMMMMM
MMMMMMMMMMWXKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK0KNMMMMMMMMWXKKKKKKXWMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMW0xkXWMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMW0c.  'dXMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMN0OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOxc.      ,dOOOOOOOOOOOOOOOOOOOO0NMMMMM
MMMMNl.                                                                  .lNMMMM
MMMMO.                                     .                              .kMMMM
MMMNl                                  ..:k0k:.                            lNMMM
MMMK;                                .:OXWMMMWk;.                          ;KMMM
MMMN0kkkkkkkkkkkkkkkkkkkkkkkkkkkkxkkk0NMMMMMMMMN0kkkkkkkkkkkkkkkkkkkkkkkkkk0NMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMN00XWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMNk;..'dNMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMWXXXXXXXXXXXXXXXXXXXXXXXKk:.     ,xKXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXNMMMM
MMMMk,.......................          ...................................,kMMMM
MMMMK,                                                                    ,KMMMM
MMMMWx.                    .cdo,                                         .xWMMMM
MMMMMNl                  .lKWMMXd'                                       lNMMMMM
MMMMMMNxlllllllllllllllldKWMMMMMMXxlllllllllllllllllllllllllllllllllllllxXMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMWXkxKWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMXd'  .lKWMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMNOo,      .ok000000000000000000000000000000000000000KNMMMMMMMMMMMMM
MMMMMMMMMMMMMNd'                                               .,xNMMMMMMMMMMMMM
MMMMMMMMMMMMMMWXd,.                                           ,dXWMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMNOl'.                                    .'lONMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMN0d:.                              .:oONMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMWX0dl;'.                  .';ldOXWMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMWNK0kxdollccccllodxk0KNWMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
                        ___  ______ _____ _____ _____
                       / _ \ | ___ \_   _|  _  /  ___|
                      / /_\ \| |_/ / | | | | | \ `--.
                      |  _  ||  __/  | | | | | |`--. \
                      | | | || |     | | \ \_/ /\__/ /
                      \_| |_/\_|     \_/  \___/\____/
```

but I can just put that in the test plan :P

**Comment by perryjrandall** [comment]
Made this a subcommand instead of command

File: crates/aptos/src/ascii/mod.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    Logo,
+}
+
+impl AsciiTool {
+    pub async fn execute(self) -> CliResult {
+        match self {
+            AsciiTool::Logo => self.logo(),
+        }
+    }
+
```
Comment (gregnazario) [inline crates/aptos/src/ascii/mod.rs:24]: This trait is consistent across all the commands and ensures logging etc is automatically handled https://github.com/aptos-labs/aptos-core/blob/main/crates/aptos/src/account/create.rs#L28-L41

**Pull request closed without merging**